app/src/main/java/org/schabi/newpipe/util/StreamDialogEntry.java

```diff
@@ -206,4 +229,30 @@ private static void openChannelFragment(final Fragment fragment,
                 fragment.requireActivity().getSupportFragmentManager(),
                 item.getServiceId(), uploaderUrl, item.getUploaderName());
     }
+
+    private static boolean isNetworkAvailable(final Context context) {
+        final ConnectivityManager cm =
+                (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
+
+        final NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
+        return activeNetwork != null
+                && activeNetwork.isConnectedOrConnecting();
+    }
+
+
+    /////////////////////////////////////////////////////////
+    // private method to show unavailable network Snackbar //
+    /////////////////////////////////////////////////////////
+
+    private static void showNetworkUnavailableSnackbar(final Fragment fragment) {
+        final Snackbar snackbar = Snackbar.make(
+                fragment.requireView(),
+                R.string.error_no_network,
+                Snackbar.LENGTH_INDEFINITE);
+        snackbar.setAction(R.string.error_no_network_action, view ->
+                NavigationHelper.openDownloads(fragment.requireActivity()));
+        // Snackbar needs to adjust its position if the system has gesture navigation
+        snackbar.setGestureInsetBottomIgnored(true);
```

Comment:
Mmmh, we aren't doing this with all other `Snackbar`s across the app, why do it here?